components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/core/OAuthParRequestWrapper.java

```diff
@@ -43,32 +42,24 @@ public class OAuthParRequestWrapper extends HttpServletRequestWrapper {
      * Wraps the request with parameters obtained from the PAR endpoint.
      *
      * @param request HttpServletRequest.
-     * @param parAuthService ParAuthService.
-     * @throws OAuthProblemException OAuthProblemException.
+     * @throws IdentityException IdentityException.
      */
-    public OAuthParRequestWrapper(HttpServletRequest request, ParAuthService parAuthService)
-            throws OAuthProblemException {
+    public OAuthParRequestWrapper(HttpServletRequest request) throws IdentityException {
 
         super(request);
 
-        //get only uuid from request_uri
+        // Get only uuid from request_uri.
         String requestUri = request.getParameter(OAuthConstants.OAuth20Params.REQUEST_URI);
         String uuid = requestUri.replaceFirst(ParConstants.REQUEST_URI_PREFIX, "");
 
         try {
-            if (parAuthService == null) {
-                throw new ParAuthFailureException("ParAuthService is not initialized properly");
-            }
-
-            params = parAuthService.retrieveParams(uuid,
-                    request.getParameter(OAuthConstants.OAuth20Params.CLIENT_ID));
+            params = ParAuthServiceComponentDataHolder.getInstance().getParAuthService()
```

Comment:
With the introduction of the new extension point, we can decouple this logic. let's resolve the params separately and provide a constructor that allows to pass in the params.